src/api_proxy/service_control/request_info.h

```diff
@@ -103,18 +103,19 @@ struct CheckRequestInfo : public OperationInfo {
   std::string ios_bundle_id;
 };
 
+enum CheckResponseErrorType {
+  UNKNOWN = 0,
+  API_KEY_INVALID = 1,
+  SERVICE_NOT_ACTIVATED = 2,
+  CONSUMER_BLOCKED = 3,
+  CONSUMER_ERROR = 4,
+};
+
 // Stores the information subtracted from the check response.
 struct CheckResponseInfo {
-  // If the request have a valid api key.
-  bool is_api_key_valid;
-  // If service is activated.
-  bool service_is_activated;
-  // Consumer project id
-  std::string consumer_project_id;
+  CheckResponseErrorType error_type = CheckResponseErrorType::UNKNOWN;
```

Comment:
Look like the default value would be `UNKNOWN` which doesn't fit the case of no error. 

Maybe use a unique_ptr to represent if it contains an error? Not sure about this.

Reply:
Good point, protos suffer from the same problem. Let's follow https://aip.dev/126 and change `UNKNOWN` to `ERROR_TYPE_UNSPECIFIED`.